### PR TITLE
Removed UID validation when any of the study, series or instance UIDs are same

### PR DESCRIPTION
**Pull request by arunmk-ms**
## Description
Removed UID validation when any of the UIDs (studyInstanceUid, seriesInstanceUid or sopInstanceUid) are same. 

## Related issues
Addresses [issue #112513](https://microsofthealth.visualstudio.com/Health/_workitems/edit/112513).

## Testing
Tested manualy with a dcm file that has same UIDs and also using unit and e2e tests.

### CODE REVIEW

**Review by bcarthic — APPROVED** [review]

File: src/Microsoft.Health.Dicom.Core/Features/Store/StoreDatasetValidator.cs [hunk 1]
```diff
@@ -147,21 +147,6 @@ private static void ValidateRequiredCoreTags(DicomDataset dicomDataset, string r
         string seriesInstanceUid = EnsureRequiredTagIsPresentWithValue(dicomDataset, DicomTag.SeriesInstanceUID);
         string sopInstanceUid = EnsureRequiredTagIsPresentWithValue(dicomDataset, DicomTag.SOPInstanceUID);
 
-        // Ensure the StudyInstanceUid != SeriesInstanceUid != sopInstanceUid
-        if (studyInstanceUid == seriesInstanceUid ||
-            studyInstanceUid == sopInstanceUid ||
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Core/Features/Store/StoreDatasetValidator.cs:152]: nit: you can remove getting values for `seriesInstanceUid` and `sopInstanceUid`

File: src/Microsoft.Health.Dicom.Core/Messages/Retrieve/RetrieveRequestValidator.cs [hunk 2]
```diff
@@ -44,30 +43,6 @@ private static void ValidateInstanceIdentifiersAreValid(ResourceType resourceTyp
         }
     }
 
-    private static void ValidateInstanceIdentifiersAreNotDuplicate(ResourceType resourceType, string studyInstanceUid, string seriesInstanceUid = null, string sopInstanceUid = null)
```
Comment (bcarthic) [inline src/Microsoft.Health.Dicom.Core/Messages/Retrieve/RetrieveRequestValidator.cs:47]: I am wondering, why we did on retrieve if we blocked on STOW? Do you know why?

**Review by bcarthic — APPROVED** [review]

File: src/Microsoft.Health.Dicom.Core/Messages/Retrieve/RetrieveRequestValidator.cs (hunk 2 above)
Comment (bcarthic) [reply src/Microsoft.Health.Dicom.Core/Messages/Retrieve/RetrieveRequestValidator.cs:47]: Found the PR #67. But I don't see any reason why

**Review by bcarthic — CHANGES_REQUESTED** [review]
Should we do only for v2?

**Review by bcarthic — CHANGES_REQUESTED** [review]
Should we do only for v2?

**Review by KnicKnic — DISMISSED** [review]

**Review by KnicKnic — DISMISSED** [review]

File: src/Microsoft.Health.Dicom.Core.UnitTests/Features/Store/StoreDatasetValidatorTestsV1.cs [hunk 3]
```diff
@@ -338,15 +338,20 @@ public async Task GivenNonRequiredTagNull_ExpectTagValidatedAndNoErrorProduced()
...
-    public async Task GivenDuplicatedIdentifiers_WhenValidated_ThenDatasetValidationExceptionShouldBeThrown(string firstDicomTagInString, string secondDicomTagInString)
+    public async Task GivenDuplicatedIdentifiers_WhenValidated_ThenValidationPasses(string firstDicomTagInString, string secondDicomTagInString)
     {
         DicomTag firstDicomTag = DicomTag.Parse(firstDicomTagInString);
         DicomTag secondDicomTag = DicomTag.Parse(secondDicomTagInString);
 
         string value = _dicomDataset.GetSingleValue<string>(firstDicomTag);
         _dicomDataset.AddOrUpdate(secondDicomTag, value);
 
-        await ExecuteAndValidateInvalidTagEntries(secondDicomTag);
```
[the remark this answers is not in the document]
Comment (arunmk-ms) [reply src/Microsoft.Health.Dicom.Core.UnitTests/Features/Store/StoreDatasetValidatorTestsV1.cs:349]: I have added the test in V2 as well

**Review by bcarthic — APPROVED** [review]

**Pull request merged**